coreclient/src/store/mod.rs

```diff
@@ -29,6 +29,22 @@ mod persistence;
 /// The result type of a failable [`Store`] method
 pub type StoreResult<T> = anyhow::Result<T>;
 
+#[derive(Debug)]
+pub enum AddHandleContactResult {
```

Comment:
I would also propose to move these types next to `add_contact_via_handle`, because they belong to the signature of the function and are used only there.